File: 백준/Silver/1920. 수 찾기/수 찾기.py
import sys
input = sys.stdin.readline

N = int(input())
arrN = list(map(int, input().split()))
M = int(input())
arrM = list(map(int, input().split()))

arrN.sort()

ans = [0] * M
for i in range(M):
    left = 0
    right = N-1
    while(left <= right):
        mid = (left+right)//2
        if(arrM[i]==arrN[mid]):
            ans[i] = 1
            break
        elif(arrM[i]<arrN[mid]):
            right = mid - 1
        else:
            left = mid + 1

for i in ans:
    print(i)

File: 백준/Silver/1920. 수 찾기/수 찾기.cc
#include <iostream>
#include <algorithm>

using namespace std;

int arrN[100000];
int arrM[100000];
bool check[100000] = {0,};

int main(){
  ios_base::sync_with_stdio(false);
  cin.tie(NULL);
  
  int N, M;
  cin >> N;
  for(int i=0; i<N; i++){
    cin >> arrN[i];
  }
  sort(arrN, arrN+N);

  cin >> M;
  for(int i=0; i<M; i++){
    cin >> arrM[i];
  }
  
  for(int i=0; i<M; i++){
    int left = 0;
    int right = N-1;

    while(left <= right){
      int mid = (left + right) / 2;
      if(arrM[i]==arrN[mid]){
        check[i] = true;
        break;
      }
      else if(arrM[i]>arrN[mid]){
        left = mid + 1;
      }
      else if(arrM[i]<arrN[mid]){
        right = mid - 1;
      }
    }
  }
  for(int i=0; i<M; i++){
    cout << check[i] << ' ';
  }
}